cpp/demo/hyperelasticity/main.cpp

```diff
@@ -170,15 +170,15 @@ int main(int argc, char* argv[])
       MPI_COMM_WORLD, pt, {{8, 8, 8}}, mesh::CellType::Type::tetrahedron,
       mesh::GhostMode::none));
 
-  auto space = std::unique_ptr<ufc_function_space, decltype(free)*>(
-      hyperelasticity_functionspace_create(), free);
-
-  auto ufc_map = std::shared_ptr<ufc_dofmap>(space->dofmap(), free);
+  ufc_function_space* space = hyperelasticity_functionspace_create();
+  ufc_dofmap* ufc_map = space->dofmap();
   auto V = std::make_shared<function::FunctionSpace>(
       mesh,
       std::make_shared<fem::FiniteElement>(
           std::shared_ptr<ufc_finite_element>(space->element(), free)),
       std::make_shared<fem::DofMap>(*ufc_map, *mesh));
+  std::free(ufc_map);
```

Comment:
Is this correct? Does `space->dofmap()` create a dofmap, or just point to the dofmap owned by `space`?

Reply:
It creates it. I guess it should have create in the name, but that's an ffc/ufc issue.